RocketControlUnitGUI/src/lib/hooks/useModals.ts

```diff
@@ -0,0 +1,128 @@
+import { getModalStore, type ModalSettings } from '@skeletonlabs/skeleton';
+import type { PocketbaseHook } from './usePocketbase';
+
+const stateToCommand: { [key: string]: string } = {
+	RS_ABORT: 'RSC_ANY_TO_ABORT',
+	RS_PRELAUNCH: 'RSC_GOTO_PRELAUNCH',
+	RS_FILL: 'RSC_GOTO_FILL',
+	RS_ARM: 'RSC_GOTO_ARM',
+	RS_IGNITION: 'RSC_GOTO_IGNITION',
+	RS_LAUNCH: 'RSC_IGNITION_TO_LAUNCH',
+	RS_BURN: 'RSC_GOTO_BURN',
+	RS_COAST: 'RSC_GOTO_COAST',
+	RS_DESCENT: 'RSC_GOTO_DESCENT',
+	RS_RECOVERY: 'RSC_GOTO_RECOVERY',
+	RS_TEST: 'RSC_GOTO_TEST'
+};
+
+const commandToState = Object.fromEntries(
+	Object.entries(stateToCommand).map(([key, value]) => [value, key])
+);
+
+Object.freeze(stateToCommand);
+Object.freeze(commandToState);
+
+export const useModals = (pocketbaseHook: PocketbaseHook) => {
+	const modalStore = getModalStore();
+
+	let nextStatePending: string = '';
+
+	const confirmStateChange = (state: string) => {
+		nextStatePending = state;
+
+		const modal: ModalSettings = {
+			type: 'confirm',
+			title: 'Please Confirm',
+			body: `Are you sure you wish to proceed to ${commandToState[state]}?`,
+			response: (r: boolean) => {
+				if (r) {
+					pocketbaseHook.writeStateChange(nextStatePending);
+				}
+
+				nextStatePending = '';
+			}
+		};
+
+		modalStore.trigger(modal);
+	};
+
+	const instantStateChange = (state: string) => {
+		nextStatePending = state;
+		pocketbaseHook.writeStateChange(nextStatePending);
+		nextStatePending = '';
+	};
```

Comment:
This one isn't technically a modal, should this stay here?

Reply:
It is not, but it needs the internal state of the hook. I renamed the hook to useInteraction.